### ISSUE TITLE
Stopped working on debian stable

### ISSUE THREAD

**Issue by mdosch** [mstilkerich/rcmcarddav#392]
Thank you very much for this handy tool. But it seems it broke on the weekly update from git yesterday. In the logs I see the following:

```
[31-Aug-2022 12:14:02 Europe/Berlin] PHP Fatal error:  Uncaught Error: Interface 'MStilkerich\CardDavAddressbook4Roundcube\Frontend\RcmInterface' not found in /var/lib/roundcube/plugins/carddav/carddav.php:39
Stack trace:
#0 /usr/share/roundcube/program/lib/Roundcube/rcube_plugin_api.php(190): include()
#1 /usr/share/roundcube/program/lib/Roundcube/rcube_plugin_api.php(121): rcube_plugin_api->load_plugin('carddav')
#2 /usr/share/roundcube/program/include/rcmail.php(100): rcube_plugin_api->load_plugins(Array, Array)
#3 /usr/share/roundcube/program/include/rcmail.php(78): rcmail->startup()
#4 /usr/share/roundcube/index.php(43): rcmail::get_instance(0, NULL)
#5 {main}
  thrown in /var/lib/roundcube/plugins/carddav/carddav.php on line 39
```

**Comment by mdosch** [comment]
I removed everything and installed it the composer way. Now it works again but the command for following stable releases only does not work so I had to use `require --update-no-dev -o "roundcube/carddav:*"`

**Comment by mstilkerich** [comment]
Hello,

when using the git master, you should be aware that you are using a development line that is less stable than released versions (although I try to keep it stable). Within the last week, I merged larger code changes that will become the next major version of this plugin to master. Please mind that after upgrading to v5, there is no easy way of going back to v4 without deleting all of rcmcarddav's database tables, as the database schema has changed for v5 and the v4 tables will be automatically migrated upon first login. Because of the error you experienced, this migration was apparently not done yet (otherwise your v4 installation would not work anymore).

That being said, I believe the issue you were facing is not a defect of rcmcarddav. The class that cannot be load is present in the current git master: https://github.com/mstilkerich/rcmcarddav/blob/master/src/Frontend/RcmInterface.php
So there must be a problem with the way you installed or updated rcmcarddav. Since you provided no detailed information on how you installed and updated rcmcarddav, I cannot give you any more detailed diagnostics of what could have gone wrong.

Installation with the packaged roundcube version in debian is a bit tricky because debian separates the roundcube installation into a part in /usr and a part in /var. The composer file resides in /usr, whereas additional stuff should be installed in /var or needs to be symlinked manually. Therefore, in the install instructions I suggested to use tarball releases with the roundcube packages in debian.

As to your last point, the composer command you showed as working is the one to install the latest stable release, so I'm not sure which command for "following the stable releases" you tried that did not work. If you need further assistance, please provide more information on what exactly is the issue you are experiencing, in what directory you ran composer and which composer invocation excactly you used to trigger the problem. If you do not need further assistance, please close the issue or I will do so in a couple of days in case of no reply.

**Comment by mdosch** [comment]
I just used a script via cron to update the plugin. Maybe it's nonsense, but somehow it worked well for almost three years:

```
cd /usr/local/share/roundcube/plugins/carddav/ || exit
sudo -u www-data git pull
sudo -u www-data composer update --no-dev
sudo -u www-data composer install --no-dev
```
Anyway, the composer-only-method seems to work and it's up and running again. :)

```
cd /var/lib/roundcube/
sudo -u www-data composer update --no-dev -o roundcube/carddav
```

As there is another command mentioned for stable releases at https://github.com/mstilkerich/rcmcarddav/blob/master/doc/INSTALL.md?plain=1#L34 I assumed the different syntax (compare the `:` means stable releases. Maybe it's just a typo but I am not very familiar with composer and it's syntax. :)

**Comment by mstilkerich** [comment]
Ok well that line in the doc just refers to the version constraint in the line before (which shows the full command including the version constraint for stable releases). The next bullet below then shows the version constraint to use instead for the git master branch. The version constraint is not a command by itself.

Install stable releases:
`php composer.phar require --update-no-dev -o "roundcube/carddav:*"`

Install git master:
`php composer.phar require --update-no-dev -o "roundcube/carddav:dev-master"`

Although I just noted that the way I wrote it is misleading as I used the syntax from the composer.json file, not the one used on the command line (e.g. `"roundcube/carddav": "*"` instead of `"roundcube/carddav:*"`.
